### DISCUSSION TITLE
[Feat] Playground

### DISCUSSION THREAD

**Discussion by ManishBisht777** [FrontendFreaks/Official-Website#110] [Ideas]
### Description

Add a playground for a better experience while learning HTML, CSS, and Javascript

**Comment by vatan999** [comment]
can you explain it in little bit more detail.

**Comment by ManishBisht777** [comment]
Actually, it is an initial thought of having an editor or something like that to test the code directly without moving back and forth from site to editor, need som more thought on this though

**Comment by Huzaifa785** [comment]
Hey!

What's coming to my mind is, that we can add the code editor on the right side of the documentation page and make the left sidebar collapsable, so that we get more space.

I think maybe we can use the `react-codemirror` package for integrating the code editor, will think more about how to implement it.

**Reply by Vishal-raj-1** [reply]
I can suggest using `react-codemirror` for playground. I have create my own coding editor, refer this repo: https://github.com/Vishal-raj-1/code_deck

Live Preview: http://code-deck.vercel.app/

**Comment by Vishal-raj-1** [comment]
Hey @Huzaifa785 Yes, We are planning to use `react-codemirror` package for code editor. Do you want to give it a try?

Code Editor will be the fourth tab

![image](https://github.com/FrontendFreaks/Official-Website/assets/59874304/45f03e33-f5b1-4417-a84d-c86f69ae853d)

**Comment by ManishBisht777** [comment]
It would be a little hassle to put the code editor in there because in each page we have to mount  a new code editor instead
- we can make a single code editor page
- which would have 2 section (like leetcode)
    -  sidebar for the task you are going to attempt
    - code editor
   
When on the Assignments tab  a  button for try now should be there which fill the code editor with
- task
- initial boiler plate for that task

**Reply by Huzaifa785** [reply]
I'll give it a try like how @ManishBisht777 described it. That would be better I guess.

**Comment by Kiranism** [comment]
![image](https://github.com/FrontendFreaks/Official-Website/assets/68379183/90162ffc-586a-4e86-a831-1d754ef84797)

CodePen-like UI; I used an iframe to showcase the results. @ManishBisht777  , is this what you meant?

**Reply by ManishBisht777** [reply]
yeah something like this but i think @Huzaifa785 was also working on this so you can collaborate together to come up with something really cool :)

**Reply by Vishal-raj-1** [reply]
Hey @Kiranism iframe is good, but there are lots of problem in simple iframe (like spacing, auto correction, auto closing bracket, and many more basic editor functionalities are missing)

I can suggest using `react-codemirror` for playground. I have create my own coding editor, refer this repo: https://github.com/Vishal-raj-1/code_deck

Live Preview: http://code-deck.vercel.app/

**Reply by Kiranism** [reply]
Hy @Vishal-raj-1 the editor that used is `react-codemirror` , which support options like autoclosing and all. The output visible in the screen is with the help of iframe as we injecting html css and js code.

**Comment by sridharrajaram** [comment]
@ManishBisht777 Issue Inside Learn >Semantic Tags notes are missing #169.
Its merged. can you solve this open issue.
